### ISSUE TITLE
Error: Failed to determine user's tenant!

### ISSUE THREAD

**Issue by tomgreenfield** [adaptlearning/adapt_authoring#2504]
## Problem

The domain session code has gone awry. The culprit this time seems to be the 5.9.x line of the Mongoose package.

## Temporary workround
```console
npm install mongoose@5.8.13
```

## Stack trace
```console
Error: Failed to determine user's tenant!
    at Object.getDatabase (D:\_delete_adapt_authoring\lib\database.js:416:19)
    at D:\_delete_adapt_authoring\plugins\content\tag\index.js:76:16
    at D:\_delete_adapt_authoring\lib\rest.js:115:7
    at Layer.handle [as handle_request] (D:\_delete_adapt_authoring\node_modules\express\lib\router\layer.js:95:5)
    at next (D:\_delete_adapt_authoring\node_modules\express\lib\router\route.js:137:13)
    at Route.dispatch (D:\_delete_adapt_authoring\node_modules\express\lib\router\route.js:112:3)
    at Layer.handle [as handle_request] (D:\_delete_adapt_authoring\node_modules\express\lib\router\layer.js:95:5)
    at D:\_delete_adapt_authoring\node_modules\express\lib\router\index.js:281:22
    at Function.process_params (D:\_delete_adapt_authoring\node_modules\express\lib\router\index.js:335:12)
    at next (D:\_delete_adapt_authoring\node_modules\express\lib\router\index.js:275:10)
    at Layer.handle [as handle_request] (D:\_delete_adapt_authoring\node_modules\express\lib\router\layer.js:91:12)
    at trim_prefix (D:\_delete_adapt_authoring\node_modules\express\lib\router\index.js:317:13)
    at D:\_delete_adapt_authoring\node_modules\express\lib\router\index.js:284:7
    at Function.process_params (D:\_delete_adapt_authoring\node_modules\express\lib\router\index.js:335:12)
    at next (D:\_delete_adapt_authoring\node_modules\express\lib\router\index.js:275:10)
    at D:\_delete_adapt_authoring\lib\permissions.js:495:16
    at Layer.handle [as handle_request] (D:\_delete_adapt_authoring\node_modules\express\lib\router\layer.js:95:5)
    at trim_prefix (D:\_delete_adapt_authoring\node_modules\express\lib\router\index.js:317:13)
    at D:\_delete_adapt_authoring\node_modules\express\lib\router\index.js:284:7
    at Function.process_params (D:\_delete_adapt_authoring\node_modules\express\lib\router\index.js:335:12)
    at next (D:\_delete_adapt_authoring\node_modules\express\lib\router\index.js:275:10)
    at SendStream.error (D:\_delete_adapt_authoring\node_modules\serve-static\index.js:121:7)
    at SendStream.emit (events.js:210:5)
    at SendStream.EventEmitter.emit (domain.js:476:20)
    at SendStream.error (D:\_delete_adapt_authoring\node_modules\send\index.js:270:17)
    at SendStream.onStatError (D:\_delete_adapt_authoring\node_modules\send\index.js:421:12)
    at next (D:\_delete_adapt_authoring\node_modules\send\index.js:735:16)
    at onstat (D:\_delete_adapt_authoring\node_modules\send\index.js:724:14)
```

## Versions
  - Authoring Tool Version: 0.10.2
  - Node.js Version: 12.11.1
  - MongoDB Version: 3.2.21
  - Operating System: Windows 10 Pro
  - Browser: Chrome 79

**Comment by kdta91** [comment]
Thanks. This does fix the problem.

**Comment by tomgreenfield** [comment]
Patched for the 0.10.3 release. Leaving this issue open for investigation into a proper fix.

**Comment by giandre** [comment]
so I have this identical issue 4 years later. Is this "workaround" still the solution?
